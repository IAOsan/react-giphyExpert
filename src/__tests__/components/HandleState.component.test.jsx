import { vi } from 'vitest';
import '@testing-library/jest-dom';
import { render } from '../test-utils';
import HandleState from '../../components/common/HandleState.component';

function renderComponent(props = {}, child) {
	return render(<HandleState {...props}>{child}</HandleState>);
}

describe('<HandleState />', () => {
	const mockLoadingComponent = <mock-loading />;
	const mockEmptyComponent = <mock-empty />;
	const mockErrorComponent = <mock-error />;
	const mockChildComponent = <mock-child />;

	it('should render loading component provided if isLoading', () => {
		const wrapper = renderComponent(
			{
				isLoading: true,
				config: {
					loadingStateElement: mockLoadingComponent,
				},
			},
			mockChildComponent
		);

		expect(wrapper.container.firstElementChild).toMatchSnapshot();
	});
	it('should render loading component provided if isEmpty', () => {
		const wrapper = renderComponent(
			{
				isEmpty: true,
				config: {
					emptyStateElement: mockEmptyComponent,
				},
			},
			mockChildComponent
		);

		expect(wrapper.container.firstElementChild).toMatchSnapshot();
	});
	it('should render loading component provided if have error', () => {
		const wrapper = renderComponent(
			{
				error: true,
				config: {
					errorStateElement: mockErrorComponent,
				},
			},
			mockChildComponent
		);

		expect(wrapper.container.firstElementChild).toMatchSnapshot();
	});
	it('should render child component if no loading, empty or error state', () => {
		const wrapper = renderComponent(mockChildComponent);

		expect(wrapper.container.firstElementChild).toMatchSnapshot();
	});
});
