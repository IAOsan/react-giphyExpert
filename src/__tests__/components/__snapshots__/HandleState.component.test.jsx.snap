// Vitest Snapshot v1

exports[`<HandleState /> > should render child component if no loading, empty or error state 1`] = `null`;

exports[`<HandleState /> > should render loading component provided if have error 1`] = `<mock-error />`;

exports[`<HandleState /> > should render loading component provided if isEmpty 1`] = `<mock-empty />`;

exports[`<HandleState /> > should render loading component provided if isLoading 1`] = `<mock-loading />`;
